File: open_notebook/domain/news.py
from typing import ClassVar, Optional

from pydantic import Field

from open_notebook.domain.base import ObjectModel


class NewsArticle(ObjectModel):
    table_name: ClassVar[str] = "news_article"

    category: str
    source: str
    title: str
    summary: str
    image: Optional[str] = None
    url: str
    time: str # Store as string for display like '2 hours ago', or we can store published date
    published_date: Optional[str] = None
    likes: int = 0
    dislikes: int = 0

File: frontend/src/lib/api/news.ts
import { apiClient } from './client'

export interface NewsArticle {
  id: string
  category: string
  source: string
  title: string
  summary: string
  image: string | null
  url: string
  time: string
  published_date: string | null
  likes: number
  dislikes: number
}

export const newsApi = {
  getNews: async (): Promise<NewsArticle[]> => {
    const { data } = await apiClient.get<NewsArticle[]>('/news')
    return data
  },
}
